mozjs-sys/build.rs

```diff
@@ -57,17 +57,30 @@ fn main() {
     // Used by mozjs downstream, don't remove.
     println!("cargo:outdir={}", build_dir.display());
 
-    if let Some(archive) = env::var_os("MOZJS_ARCHIVE") {
-        download_static_lib_binaries(&PathBuf::from(archive), &build_dir);
+    // Link to pre-built archive first if it exists.
+    let create_archive = env::var_os("MOZJS_CREATE_ARCHIVE").is_some();
+    let build_from_source = if env::var_os("MOZJS_FROM_SOURCE").is_some() || create_archive {
+        true
```

Comment:
If I am reading things correctly, if features mismatch with what artifacts provide we run compilation with artifacts anyway and then on failure (if it even fails), we do fallback. I think it would be nicer to fallback to build from source directly. So if `streams` feature is not selected or `debug-mozjs` feature is enabled. And when building from source is enabled we should print reason for this using `cargo:info` for easier debugging.

Reply:
I added two conditions to check streams and debug-mozjs features. Do they match what you described?

Reply:
It's good. Is there any specific reason for mixing `env::var` vs `env::var_os`, or can we only use one variant everywhere?

Separating `MOZJS_FROM_SOURCE` and `MOZJS_CREATE_ARCHIVE` into separate if's with reasons info would also make sense, right?

Otherwise this looks really good, I will do more test after #459 to make sure everything works as expected and then we can land this.